iroha/src/lib.rs

```diff
@@ -211,28 +236,24 @@ where
     /// # Errors
     /// Can fail if initing kura fails
     #[iroha_futures::telemetry_future]
-    pub async fn start(&mut self) -> Result<Infallible> {
+    pub async fn start(&mut self) {
         iroha_logger::info!("Starting Iroha.");
-        self.torii
-            .take()
-            .ok_or_else(|| error!("Seems like peer was already started"))?
-            .start()
-            .await
-            .wrap_err("Failed to start Torii")
+        match self.torii.take() {
+            None => iroha_logger::error!("Seems like peer was already started"),
+            Some(torii) => torii.start().await,
+        }
     }
```

Comment:
Can we still return `Result` here? To signal that peer was already started before

Reply:
It's impossible to construct infallible. Is it okay here to use `Result<()>`?

Reply:
Yes! The point of infallible is to show that function wont terminate, but here i guess as we cant change warp's signatures we can't also use Infallible

Reply:
Fixed.